### ISSUE TITLE
SC2329 should be transitive

### ISSUE THREAD

**Issue by ensc** [koalaman/shellcheck#3054]
#### For bugs
- Rule id SC2329
- My shellcheck version (`shellcheck --version` or "online"): online;  0.9.0 shows similar results but for SC2317
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```bash
#! /bin/bash

note() {
    echo "$*"
}

# shellcheck disable=SC2317,SC2329
tool_a() {
    note a
}

# shellcheck disable=SC2317,SC2329
tool_b() {
    echo b
}

RC=0

for i in a b; do
    tool_"$i" || RC=1
done

exit $RC
```

#### Here's what shellcheck currently says:

```
[Line 3:](javascript:setPosition(3, 1))
note() {
^-- [SC2329](https://www.shellcheck.net/wiki/SC2329) (info): This function is never invoked. Check usage (or ignored if invoked indirectly).
```

#### Here's what I wanted or expected to see:

no warnings;

SC2329 should not be triggered for functions which are called by functions with disabled SC2329 check

**Comment by wileyhy** [comment]
Hm. I get a different result on a Fedora 40 local vm instance: SC2317 on
line 4, 'echo "$*"', "(info): Command appears unreachable...." The rpm is
'ShellCheck-0.9.0-6.fc40.x86_64'. The same output as yours on shellcheck.net,
though.

Wiley

**Comment by ensc** [comment]
> Hm. I get a different result on a Fedora 40 local vm instance: SC2317 on line 4

SC2329 seems to have been splitted off from SC2317 in recent versions (used on http://spellcheck.net, but not (yet) by Fedora 40)

**Comment by koalaman** [comment]
This should be working as described on the latest git commit, and also on shellcheck.net (which may not have been published when this issue was filed).

**Comment by MichaIng** [comment]
I am confused that SC2329 was in some v0.9.0 commit, while it was now officially added with v0.11.0? However, with v0.11.0 and on current https://www.shellcheck.net/, the code throws the warning (again):
```console
[Line 3:](javascript:setPosition(3, 1))
note() {
^-- [SC2329](https://www.shellcheck.net/wiki/SC2329) (info): This function is never invoked. Check usage (or ignored if invoked indirectly).
```
